nimble/transport/apollo3/src/apollo3_ble_hci.c

```diff
@@ -275,16 +255,22 @@ apollo3_ble_hci_acl_tx(struct os_mbuf *om)
 {
     struct os_mbuf *x;
     int rc = 0;
+    hci_drv_write_t write_buf;
+    uint8_t *ptr = &write_buf.data[1];
+    
+    write_buf.data[0] = HCI_H4_ACL;
+    write_buf.len = 1;
 
     x = om;
     while (x) {
-        rc = apollo3_hci_write(HCI_H4_ACL, x->om_len, x->om_data);
-        if (rc < 0) {
-            break;
-        }
+        memcpy(ptr, x->om_data, x->om_len);
```

Comment:
Is there a way to determine the length to pass into `os_mbuf_copydata`? If we grab it by traversing and summing up the individual mbuf om->len values then it doesn't eliminate the loop

Reply:
I just noticed (not being expert in mbuf field) that there is a function that copies data to flat buffer.
There is also os_mbuf_len() function to get data len (or OS_MBUF_PKTLEN although I don't know if it could be applied here).
I don't mind either way.

Reply:
you can use `OS_MBUF_PKTLEN` here since all ACL data going to controller should be allocated with pkthdr